pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,218 @@
+    scalar-valued function of the hamiltonian. Calling the ``ParametrizedHamiltonian`` returns an
+    ``Operator`` representing an instance of the Hamiltonian with the specified parameter values.
+
+    .. note::
+
+        The parameters used in the ``ParametrizedHamiltonian`` call should have the same order
+        as the functions used to define this hamiltonian. For example, if we build a
+        ``ParametrizedHamiltonian`` that contains two functions:
+
+        >>> import jax.numpy as jnp
+        >>> def f1(p, t):
+        ...     return p * t
+        >>> def f2(p, t):
+        ...     return p * jnp.sin(t)
+        >>> coeffs = [f1, f2]
+        >>> ops = [qml.PauliX(0), qml.PauliY(1)]
+        >>> H = qml.ops.dot(coeffs, ops)
+
+        And we call it using the following parameters:
+
+        >>> params = [4, 5]
+        >>> H(params, t=0.5)
+        (2.0*(PauliX(wires=[0]))) + (2.397127628326416*(PauliY(wires=[1])))
+
+        Internally we are computing ``f1(4, 0.5)`` and ``f2(5, 0.5)``.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        coeffs = [2, lambda v, t: v[0] * jnp.sin(v[1] * t)]
+        observables =  [qml.PauliX(0), qml.PauliY(1)]
+        H = ParametrizedHamiltonian(coeffs, observables)
+
+    >>> H([jnp.ones(2)], 1.)
+    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1])))
+
+    A ``ParametrizedHamiltonian`` can be created by passing a list of coefficients (scalars or functions), as well as
+    a list of corresponding observables. The functions must have two arguments, the first one being the
+    trainable parameters and the second one being time.
+
+    >>> def f1(p, t): return np.sin(p * t)
+    >>> def f2(p, t): return p * np.cos(t)
+
+    The functions, along with scalar coefficients, can then be used to initialize a ``ParametrizedHamiltonian``:
+
+    .. code-block:: python3
+
+        coeffs = [2, f1, f2]
+        obs = [qml.PauliX(0) @ qml.PauliX(1), qml.PauliY(0) @ qml.PauliY(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+        H = ParametrizedHamiltonian(coeffs, obs)
+
+    The resulting object can be passed parameters, and will return an ``Operator`` representing the
+    ``ParametrizedHamiltonian`` with the specified parameters:
+
+    >>> H([1.2, 2.3], 0.5)
+    (2*(PauliX(wires=[0]) @ PauliX(wires=[1]))) + ((0.5646424733950354*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (2.0184398923478573*(PauliZ(wires=[0]) @ PauliZ(wires=[1]))))
+
+    We can also access the fixed and parametrized terms of the ``ParametrizedHamiltonian``.
+    The fixed term is an ``Operator``, while the parametrized term must be initialized with concrete
+    parameters to obtain an ``Operator``:
+
+    >>> H.H_fixed()
+    2*(PauliX(wires=[0]) @ PauliX(wires=[1]))
+    >>> H.H_parametrized([2.5, 3.6], 0.5)
+    (0.9489846193555862*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (3.159297222805342*(PauliZ(wires=[0]) @ PauliZ(wires=[1])))
+    """
```

Comment:
Not required now, but moving forward I would add a section on Usage details going a bit deeper into the call structure and especially how that works when re-combining (parametrized) Hamiltonians

Reply:
All the examples here are directly calling `qml.ops.ParametrizedHamiltonian(coeffs, ops)`. If this isn't the intended user interface, but rather something like 

```
coeffs = [2, f1, f2]
ops = [XX, YY, ZZ]
H = qml.ops.dot(coeffs, ops)
```

or

```
f1 = lambda param, t: param[0] * np.sin(t) * (t - 1)
f2 = lambda param, t: param[1] * np.cos(t ** 2)

XX = qml.PauliX(0) @ qml.PauliX(1)
YY = qml.PauliY(0) @ qml.PauliY(1)
ZZ = qml.PauliZ(0) @ qml.PauliZ(1)

H = 2 * XX + f1 * YY + f2 * ZZ
```

should these be included and emphasized?

Reply:
Good point! Since this is quite a cool feature I'd definitely include it in the doc string! And yes I'd change it to the dot constructor :)

Reply:
Usage details should also note that the order of the coefficients is expected to be first fixed and then parametrized coefficients (otherwise wire order is not consistent between input order and everywhere else). But I don't think I have time to write this section this evening.

Reply:
I would even put this in the main docs!

Reply:
Can we add any relevant notes to the story on unifying and cleaning docs and mark this as resolved for now?

Reply:
Yes that sounds good!

Reply:
added it to the unify docs story and marking as resolved